graphql/achievement/mutations.ts

```diff
@@ -0,0 +1,34 @@
+import { Arg, Authorized, Ctx, Mutation, Resolver } from 'type-graphql';
+import { User_achievement as Achievement } from '../generated';
+import { Role } from '../roles';
+import * as Notification from '../../common/notification';
+import { GraphQLContext } from '../context';
+import { AuthorizedDeferred, hasAccess } from '../authorizations';
+import { prisma } from '../../common/prisma';
+
+@Resolver(() => Achievement)
+export class MutateAchievementResolver {
+    @Mutation((returns) => Boolean)
+    @AuthorizedDeferred(Role.ADMIN, Role.OWNER)
+    async subcourseMeetingJoin(@Ctx() context: GraphQLContext, @Arg('subcourseId') subcourseId: number) {
+        const { user } = context;
+        const subcourse = await prisma.subcourse.findUnique({ where: { id: subcourseId }, include: { course: true, lecture: true } });
```

Comment:
What if there is no subcourse found?

Reply:
The findUnique was replaced with findUniqueOrThrow.